build-logic/src/main/kotlin/publishing/configurePom.kt

```diff
@@ -103,7 +103,10 @@ internal fun configurePom(project: Project, mavenPublication: MavenPublication,
               connection.set("scm:git:$codeRepo")
               developerConnection.set("scm:git:$codeRepo")
               url.set("$codeRepo/tree/main")
-              tag.set("main")
+              val version = project.version.toString()
+              if (!version.endsWith("-SNAPSHOT")) {
+                tag.set("apache-polaris-$version")
```

Comment:
Q: Does it work automatically when we retag with the same version(e.g., `1.1.0`)? Or we need to remove the tag first, and run the command `mvn release:prepare` to retag it?

Retag could be useful when we have multiple release candidates.

Reply:
@flyrain this has nothing to do with Git. It's the project version of the build.
Changing an existing Git release tag is really bad practice.
BTW: we use Gradle, not Maven.

Reply:
I guess this is more of a release process question: For RCs are we going to publish each RC to Maven, or just the final release? If the former case, I suppose tags should be something line `apache-polaris-0.10.0-rc1`, etc. WDYT?

Reply:
I think the RCs are not published to maven repository. Taking this as an example, https://mvnrepository.com/artifact/org.apache.iceberg/iceberg-core, we can only see the formal ones, like 1.8.0, 1.8.1. 

> @flyrain this has nothing to do with Git. It's the project version of the build.

@snazy I'm confused by this statement, what do you mean by "nothing related to git"? What does it tag in that case?

Reply:
It's about Maven staging and content of the pom.

All releases (RC) are published to Maven staging repository, and if the RC pass the staging repo is promoted to Maven Central. If the RC doesn't pass, them the staging repository is deleted.
That's why it looks only the "final" release are on Maven Central, but actually all RCs went to Maven Staging.

The process is:
- you stage a release on repository.apache.org as a staging repository
- if release fails, staging repository is dropped
- if release pass, staging repository is promoted to Maven Central.

Reply:
but promoting to Maven Central does not change POM files, right? So POM should contain the "final" tag even while the release is in progress, it seems :thinking:

Reply:
That's correct: the RC should contain the "final" tag (and the branch the next SNAPSHOT).